Draft: Scroll driven animation properties and types (#41765)
mdn/content: files/en-us/web/css/animation-timeline/view/index.md

---

The **`view()`** [CSS function](/en-US/docs/Web/CSS/CSS_values_and_units/CSS_value_functions) can be used to define the axis and insets of an [anonymous view progress timeline](/en-US/docs/Web/CSS/CSS_scroll-driven_animations/Timelines#anonymous_view_progress_timelines_the_view_function).

## Syntax

```css
/* No parameters */
animation-timeline: view();

/* Axis parameter */
animation-timeline: view(block);
animation-timeline: view(x);

/* Inset parameter */
animation-timeline: view(auto);
animation-timeline: view(20%);
animation-timeline: view(200px);
animation-timeline: view(20% 40%);
animation-timeline: view(20% 200px);
animation-timeline: view(100px 200px);
animation-timeline: view(auto 200px);

/* Axis and inset parameters */
animation-timeline: view(block auto);
animation-timeline: view(inline 20%);
animation-timeline: view(x 200px auto);
```

### Parameters

- `<axis>`
  - : An {{ cssxref("axis") }} keyword value describing the direction, or axis, of the scrollport that controls the scroll-driven animation. The default value is `block`.
- `<view-timeline-inset>`
  - : Up to two {{cssxref("length-percentage")}} values or the keyword `auto`.

## Description

> The `view()` function is used with the {{cssxref("animation-timeline")}} property to indicate that the nearest ancestor scroll container will provide an anonymous view progress timeline to animate, optionally providing the direction of the scroller and starting and ending timeline insets. View progress timelines progress through by a change in visibility of subject elements inside their nearest ancestor scrollers.

Reviewer: ```suggestion
The `view()` function is used with the {{cssxref("animation-timeline")}} property to indicate that the nearest ancestor scroll container will provide an anonymous view progress timeline to animate, optionally providing the direction of the scroller and starting and ending timeline insets. View progress timelines are progressed by subject elements changing visibility inside their nearest ancestor scrollers.
```

> The function parameters can specify the scrollbar axis along which timeline progress will be tracked and an inset that adjusts the position of the box in which the subject is deemed to be visible.

Reviewer: ```suggestion
The function parameters can specify the scrollbar axis along which timeline progress will be tracked and one or two insets that adjust the position of the box in which the subject is deemed to be visible.
```

By default, the `view()` references the block axis. This can be changed by providing an explicit `<axis>` value. If the indicated axis does not contain a scrollbar, then the animation timeline will be inactive (have zero progress).

By default, the timeline is at its `0%` or `from` of its {{cssxref("keyframe")}} animation when the subject is first visible at one edge of the scroller, and at the `100%` or `to` keyframe when its outer border edge reaches the opposite edge of the scroller. This can be controlled with the `<view-timeline-inset>` parameters.

> The animation lasts as long as the element is in the inset-adjusted view.The timeline inset consists of up to two values, which can be either `auto` or a {{cssxref("length-percentage")}}. The first value is the start, which defines the inward offset from beginning of the scrollport. The second value, if present, is the end, defining the inward offset from end of the scrollport. If the value is greater than `0`, it specifies an inset (positive). A negative value defines an outset adjustment to the [scrollport](/en-US/docs/Glossary/Scroll_container#scrollport). The inset is used to determine whether the element is in view which determines the length of the animation timeline.

Reviewer: ```suggestion
The animation lasts as long as the element is in the inset-adjusted view. The timeline inset consists of up to two values, which can be either `auto` or a {{cssxref("length-percentage")}}. The first value is the start, which defines the inward offset from the [scrollport](/en-US/docs/Glossary/Scroll_container#scrollport) beginning. The second value, if present, is the end, defining the inward offset from scrollport end. If the value is greater than `0`, it specifies an inset (positive). A negative value defines an outset adjustment to the scrollport. The inset is used to determine whether the element is in view, which determines the length of the animation timeline.
```

> While the axis and inset components can be specified in any order, within the inset component, the first value defines the starting inset and the second component defines the end offset.

Reviewer: I'm not sure what you mean here. Are you trying to say that the axis and inset parameters can be specified in either order? Is that true?
